### PR TITLE
add script to harvest qnl iiif manifests

### PR DESCRIPTION
**Pull request by jacobthill**

### CODE REVIEW

File: harvest-refactor/qnl-iiif-manifest-harvest.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+from lxml import etree
+import glob
+import urllib.request
+
+ns = {'oai': "http://www.openarchives.org/OAI/2.0/"}
+
+files = glob.glob('/Users/jtim/Dropbox/DLSS/DLME/dlme-metadata/qnl/british-library/data/*.xml')
```
Comment (peetucket) [inline harvest-refactor/qnl-iiif-manifest-harvest.py:7]: Is there a way to do this as a relative path or maybe pass in the path as a parameter from the command line so you don't have to hardcode something specific to your laptop here?

**Comment by anarchivist** [comment]
I'm not entirely sure why we need to harvest the manifests, but I'm guessing I was somewhat unclear. From my Slack messages:

> So, in this case, the manifest URI you've constructed doesn't technically exist, so QDL's server is redirecting you to a different manifest URI
> I think we need a change to the macro logic to handle redirects

Looking into this a little further I think this is the file we need to change: https://github.com/sul-dlss/dlme-transform/blob/master/lib/dlme_utils.rb

**Comment by jacobthill** [comment]
We technically don’t have to harvest the manifests but it will speed things up a bit and we won’t have to hit QNL’s server 16k times every time we run a transformation

**Comment by anarchivist** [comment]
I'm not sure that we need to do the lookups here. The QNL metadata represents individual items that are sections of the manifests; the IIIF lookup mostly helps with grabbing the thumbnails, which won't be accurate (it'll just grab the first image of from the manifest, if I'm not mistaken).

**Comment by jacobthill** [comment]
That might be right. I didn't catch that before but maybe we don't even want thumbnails in that case. This is what Marcin at QDL wrote: 

> Regarding thumbnails, is URL to IIIF manifest enough to display them? If yes, then having the record identifier you can get the URL to manifest:
> 
> https://www.qdl.qa/en/iiif/<IDENTIFIER>/manifest
> 
> e.g.
> 
> https://www.qdl.qa/en/iiif/81055/vdc_100000000833.0x000360/manifest
> 
> 
I have checked half a dozen and that does seem to be the case. If that is the case then we can either include these or leave them out all together. I'm not sure QDL can do much better since some of the books only have blank covers. Should we just leave the thumbnails out or include what they provide knowing that it wont add a lot of value but may improve over time?

**Comment by anarchivist** [comment]
When you get back in touch with Marcin, I'd make sure he understands that we are looking at the item-level records that were harvested. The manifests that are linked point to larger items in which the record-level items are contained.

**Comment by jacobthill** [comment]
I sent Marcin the email explaining everything. He contacted the company that they contract for their metadata work and wrote back: 

> Cogapp will improve the OAI-PMH interface, so that the MODS records will include links to thumbnails which you need. They will also check the (A)-level IIIF manifests, as they were assuming that these manifests include (B)-level structural information as well (but is seems that they don’t).
> 

So, I think we can hold off on this until they complete this work, then re-harvest and adjust the config file as needed.

**Comment by jacobthill** [comment]
Broken out into https://github.com/sul-dlss/dlme-harvest/issues/23 and https://github.com/sul-dlss/dlme-traject/issues/78

**Pull request closed without merging**